src/back/glsl/mod.rs

```diff
@@ -2929,50 +2936,74 @@ impl<'a, W: Write> Writer<'a, W> {
+                                    }
 
-                                write!(self.out, "mix(vec{s}(31.0) - floor(log2(vec{s}(")?;
-                                self.write_expr(arg, ctx)?;
-                                write!(self.out, ") + 0.5)), ")?;
-
-                                if let crate::ScalarKind::Uint = kind {
-                                    write!(self.out, "vec{s}(32.0), lessThanEqual(")?;
                                     self.write_expr(arg, ctx)?;
-                                    write!(self.out, ", uvec{s}(0u))))")?;
-                                } else {
-                                    write!(self.out, "mix(vec{s}(0.0), vec{s}(32.0), equal(")?;
-                                    self.write_expr(arg, ctx)?;
-                                    write!(self.out, ", ivec{s}(0))), lessThanEqual(")?;
-                                    self.write_expr(arg, ctx)?;
-                                    write!(self.out, ", ivec{s}(0))))")?;
+                                    write!(self.out, "))")?;
                                 }
-                            }
-                            crate::TypeInner::Scalar { kind, .. } => {
-                                write!(self.out, "(")?;
-                                self.write_expr(arg, ctx)?;
-
-                                if let crate::ScalarKind::Uint = kind {
-                                    write!(self.out, " == 0u ? 32u : uint(")?;
-                                } else {
-                                    write!(self.out, " <= 0 ? (")?;
-                                    self.write_expr(arg, ctx)?;
-                                    write!(self.out, " == 0 ? 32 : 0) : int(")?;
+                                _ => unreachable!(),
+                            };
+                        } else {
+                            match *ctx.info[arg].ty.inner_with(&self.module.types) {
+                                crate::TypeInner::Vector { size, kind, .. } => {
+                                    let s = back::vector_size_str(size);
+
+                                    if let crate::ScalarKind::Uint = kind {
+                                        write!(self.out, "uvec{s}(")?;
+                                        write!(self.out, "vec{s}(31.0) - floor(log2(vec{s}(")?;
+                                        self.write_expr(arg, ctx)?;
+                                        write!(self.out, ") + 0.5)))")?;
+                                    } else {
+                                        write!(self.out, "ivec{s}(")?;
+                                        write!(self.out, "mix(vec{s}(31.0) - floor(log2(vec{s}(")?;
+                                        self.write_expr(arg, ctx)?;
+                                        write!(self.out, ") + 0.5)), ")?;
+                                        write!(self.out, "vec{s}(0.0), lessThanEqual(")?;
```

Comment:
```suggestion
                                        write!(self.out, "vec{s}(0.0), lessThan(")?;
```
I think this needs to be `lessThan`

Reply:
Yeah, sorry, staring at a bunch of nested parentheses in strings is started to take it’s toll 😂